src/main/java/org/apache/commons/compress/compressors/lzma/LZMACompressorInputStream.java

```diff
@@ -56,7 +56,7 @@ public LZMACompressorInputStream(final InputStream inputStream)
             in = new LZMAInputStream(inputStream, MemoryLimit.getMemoryLimitInKb());
         } catch (org.tukaani.xz.MemoryLimitException e) {
             //convert to commons-compress exception
-            throw new MemoryLimitException("exceeded calculated memory limit", e);
+            throw new MemoryLimitException(e.getMemoryNeeded(), e.getMemoryLimit());
```

Comment:
Y.  Good point.  I'll update our MemoryLimitException initializers to both a 2-parameter (needed, limit) and 3-parameter (needed, limit, e)?